forgerock-auth/src/main/java/org/forgerock/android/auth/centralize/BrowserLauncher.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.centralize
+
+import android.content.Intent
+import androidx.activity.ComponentActivity
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import net.openid.appauth.AuthorizationResponse
+import org.forgerock.android.auth.AuthorizeContract
+import org.forgerock.android.auth.FRListener
+import org.forgerock.android.auth.FRUser
+import org.forgerock.android.auth.Listener
+import java.lang.IllegalStateException
+
+/**
+ * Singleton class to launch the Browser
+ * Centralize login browser launcher.
+ */
+internal object BrowserLauncher {
+
+    private var launcher: Launcher? = null
+
+    /**
+     * Initialize the Launcher
+     */
+    @Synchronized
+    internal fun init(launcher: Launcher) {
+        BrowserLauncher.launcher = launcher
+    }
+
+    /**
+     * reset the Launcher state
+     */
+    @Synchronized
+    internal fun reset() {
+        launcher?.authorize?.unregister()
+        launcher = null
+    }
+
+    /**
+     * Authorize the user using the Browser
+     */
+    suspend fun authorize(browser: FRUser.Browser): AuthorizationResponse {
+        return launcher?.authorize(browser)
+            ?: throw IllegalStateException("Launcher is not initialized")
+    }
+
+    /**
+     * Authorize the user using the Browser
+      */
+    fun authorize(browser: FRUser.Browser, listener: FRListener<AuthorizationResponse>): Boolean {
+        return launcher?.let {
+            val scope = CoroutineScope(Dispatchers.Default)
+            scope.launch {
+                try {
+                    val result = authorize(browser)
+                    Listener.onSuccess(listener, result)
+                } catch (e: Exception) {
+                    Listener.onException(listener, e)
+                }
+            }
+            return true
+        } ?: false
+    }
+}
+
+/**
+ * Register the Browser Launcher
+ */
+fun ComponentActivity.registerBrowserLauncher() {
+    val state: MutableStateFlow<Intent?> = MutableStateFlow(null)
```

Comment:
i think if we use MutableSharedFlow<Intent>() , then you can avoid this default value i belive and dont have to drop the first value

Reply:
Some information about best practice of using Flow Type

https://medium.com/@mortitech/sharedflow-vs-stateflow-a-comprehensive-guide-to-kotlin-flows-503576b4de31

```
Here are some best practices for using SharedFlow and StateFlow in Kotlin:

Choose the right flow:
Use SharedFlow when you need to broadcast values to multiple collectors or when you want to have multiple subscribers to the same stream of data.
Use StateFlow when you need to maintain and share a single source of truth for a state and automatically update all collectors with the latest state.

```

In this case, we only have one listener and single source.